### ISSUE TITLE
PHP Fatal error:  Variadic parameter cannot have a default value

### ISSUE THREAD

**Issue by keksa** [eloquent/phony#248]
When creating a mock of the following class, phony will crash with `Variadic parameter cannot have a default value`. The problem is with the `?int ...$statuses` definition, without `?` there is no crash.

### class being mocked
```php
class Foo {
    /**
     * @var int|null
     */
    private $status;

    public function hasStatus(?int ...$statuses): bool
    {
        return in_array($this->status, $statuses, true);
    }
}
```

### phony-generated source of mock
```php
public function hasStatus(
    int ...$a0 = null
) : bool {
    $argumentCount = \func_num_args();
    $arguments = [];


    for ($i = 0; $i < $argumentCount; ++$i) {
        $arguments[] = $a0[$i - 0];
    }

    if (!$this->_handle) {
        $result = parent::hasStatus(...$arguments);

        return $result;
    }

    $result = $this->_handle->spy(__FUNCTION__)->invokeWith(
        new \Eloquent\Phony\Call\Arguments($arguments)
    );

    return $result;
}
```

**Comment by ezzatron** [comment]
Thanks for the report, I'll look into it. Which version of PHP are you running, out of interest?

**Comment by keksa** [comment]
Hi @ezzatron, I'm on PHP 7.3.21.

**Comment by ezzatron** [comment]
Fixed in [4.0.1](https://github.com/eloquent/phony/releases/4.0.1).

**Comment by keksa** [comment]
@ezzatron Thank you!
